src/Network/OAuth/OAuth2/Compat.hs

```diff
@@ -39,7 +53,13 @@ data OAuth2 = OAuth2
     , oauth2RedirectUri :: Maybe (URIRef Absolute)
     }
 
-type OAuth2Result err a = Either (OAuth2Error err) a
+#if MIN_VERSION_hoauth2(2,7,0)
+type Error = TokenRequestError
+#else
+type Error = OAuth2Error Errors
+#endif
+
+type OAuth2Result a = Either Error a
```

Comment:
Why do we have to export `Error` if we're speaking in terms of this, which kind of hides it?

Reply:
`Error` gets [rewrapped](https://github.com/freckle/yesod-auth-oauth2/blob/d8011561b8159d482b3aa300cfed5d20fc1353b7/src/Yesod/Auth/OAuth2/DispatchError.hs#L34) and potentially "re-raised" when we [try to get a token](https://github.com/freckle/yesod-auth-oauth2/blob/36805f0580bedec39a69c254456df14474f3bec1/src/Yesod/Auth/OAuth2/Dispatch.hs#L89).

Reply:
Would you recommend keeping the type parameter so it's less hidden?

Reply:
> Would you recommend keeping the type parameter so it's less hidden?

I don't mind so much about hidden, but I think keeping the type parameter allows us to avoid ourselves being a major version bump. If so, I'd recommend it for that reason.

Reply:
Another essentially internal type.